### PR TITLE
Interpret inputs refactor

### PR DESCRIPTION
**Pull request by sheaf**
This is a first step in explicitly passing input / player data instead of relying on global access to player data.

For the moment, this change makes it so that all the routines that only need input data (and not the whole player data) are passed input data and only use input data. There are still a few edge cases to sort out so occasionally the routines still access global player data, but eventually we should be able to remove the `player` import in these functions.

Ideally I think the `player` object should not have input attributes. For instance, the physics routine should be passed two things: the player objects, and the player inputs, separately. Unfortunately, doing this would require refactoring most of the codebase, changing calls like
`player[i].inputs.buttons.a[j]` to `input[j].a`, where `input` is the name of the data (of a fixed player) that is passed to the relevant functions.

No regressions on my end, please confirm @schmooblidon. (In particular, could you please check that the Z button is behaving properly everywhere, I had a few issues with it, in the target test stage builder in particular.)
@nolram88 @TJohnW Please let me know how you think we can refactor the player object to separate player data from input data. I can obviously run some regexp to replace `player[i].inputs.buttons.a[j]` to `input[j].a` (etc) wherever needed, but we need to be on the same page before doing anything drastic.

### CODE REVIEW

**Comment by nolram88** [comment]
Ok I believe  its finally ready for some real tests. The idea is everything should still work the same.

File: src/menus/gameplaymenu.js [hunk 1]
```diff
@@ -90,7 +90,7 @@ export function gameplayMenuControls (i, input){
         menuMove = true;
       }
     }
-  } else if (input[0].lsY < -0.7) {
+    } else if (player[i].inputs.lStickAxis[0].x > 0.7 && !(Math.abs(player[i].inputs.lStickAxis[0].y >= 0.7))) {
```
Comment (sheaf) [inline src/menus/gameplaymenu.js]: This should be `input[0].lsX` and `input[0].lsY`.

File: src/physics/actionStateShortcuts.js [hunk 2]
```diff
@@ -429,12 +429,27 @@ export function checkForJump (p,input){
   if ((input[p][0].x && !input[p][1].x) || (input[p][0].y && !input[p][1].y)) {
     return [true, 0];
   } else if (input[p][0].lsY > 0.66 && input[p][3].lsY < 0.2) {
+  } else if ((gameSettings["tapJumpOffp" + (p + 1)] == false || (gameMode === 4)) &&  (player[p].inputs.lStickAxis[0].y > 0.66 && player[p].inputs.lStickAxis[3].y < 0.2)) { // == is on purpose
```
Comment (sheaf) [inline src/physics/actionStateShortcuts.js]: This should also use `input[p].lsX` and `input[p].lsY`.

File: src/menus/gameplaymenu.js [hunk 3]
```diff
@@ -23,7 +23,7 @@ export function gameplayMenuControls (i, input){
   var menuMove = false;
   if (input[0].b && !input[1].b) {
     sounds.menuBack.play();
-    //input[i].b[1] = true;
+    input[i][1].b = true;
```
Comment (sheaf) [inline src/menus/gameplaymenu.js]: Can we disallow menus from modifying the input data? Pretty please?

File: src/physics/actionStateShortcuts.js [hunk 4]
```diff
@@ -429,12 +429,27 @@ export function checkForJump (p,input){
...
     return [true, 1];
   } else {
     return [false, false];
   }
 }
+export function checkForDoubleJump (p){
+  if (((player[p].inputs.x[0] && !player[p].inputs.x[1]) || (player[p].inputs.y[0] && !player[p].inputs.y[1]) || ((gameSettings["tapJumpOffp" + (p + 1)] == false || (gameMode === 4)) && player[p].inputs.lStickAxis[0].y > 0.7 && player[p].inputs.lStickAxis[1].y <= 0.7)) && (!player[p].phys.doubleJumped || (player[p].phys.jumpsUsed < 5 && player[p].charAttributes.multiJump))) {
```
Comment (sheaf) [inline src/physics/actionStateShortcuts.js]: This should use `input[p]` too.

File: src/physics/actionStateShortcuts.js [hunk 5]
```diff
@@ -429,12 +429,27 @@ export function checkForJump (p,input){
...
+    return true;
+  } else {
+    return false;
+  }
+}
+export function checkForMultiJump (p){
+  if (player[p].inputs.x[0] || player[p].inputs.y[0] || ((gameSettings["tapJumpOffp" + (p + 1)] == false || (gameMode === 4)) &&  player[p].inputs.lStickAxis[0].y > 0.7)) {
```
Comment (sheaf) [inline src/physics/actionStateShortcuts.js]: Ditto.

File: src/menus/gameplaymenu.js (hunk 3 above)
Comment (nolram88) [inline src/menus/gameplaymenu.js]: pretty good question. Im not sure of the implication here.

**Comment by nolram88** [comment]
if you want to debug get chrome or chrome canary

**Pull request merged**